src/snowflake/cli/plugins/object/stage/commands.py

```diff
@@ -20,14 +21,19 @@
 )
 
 StageNameArgument = typer.Argument(..., help="Name of the stage.")
+PatternOption = pattern_option(
+    help_example='list --pattern=".*\.txt will list all files with .txt extension'
+)
```

Comment:
Any reason why not include this in flag definition? If this is a stage operation it should be generic for stage as well as repo?